/*
   Copyright (c) 2019 AT&T Intellectual Property.
   Copyright (c) 2019 Nokia.

   Licensed under the Creative Commons Attribution 4.0 International
   Public License (the "License"); you may not use this file except
   in compliance with the License. You may obtain a copy of the License at

       https://creativecommons.org/licenses/by/4.0/

   Unless required by applicable law or agreed to in writing, documentation
   distributed under the License is distributed on an "AS IS" BASIS,
   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   See the License for the specific language governing permissions and
   limitations under the License.
*/

syntax = "proto3";

package streaming_protobufs;

option go_package = "gerrit.o-ran-sc.org/r/ric-plt/streaming-protobufs";

import "google/protobuf/wrappers.proto";
import "error_cause.proto";

/*
   Protocol Buffer definition for SGNB RECONFIGURATION COMPLETE message.
   (3GPP 36.423, ver 15.4.0)

   Naming convention follows the ASN.1 field naming defined in chapter
   '9.3 Message and Information Element Abstract Syntax (with ASN.1)' of below document:
   http://www.3gpp.org/ftp//Specs/archive/36_series/36.423/36423-f40.zip

   Dashes converted to underscores to make protobuf message syntax correct.
*/

message SgNBReconfigurationComplete
{
    uint32 id_MeNB_UE_X2AP_ID = 1;
    uint32 id_SgNB_UE_X2AP_ID = 2;
    ResponseInformationSgNBReconfComp id_ResponseInformationSgNBReconfComp = 3;
    google.protobuf.UInt32Value id_MeNB_UE_X2AP_ID_Extension = 4;
}

message ResponseInformationSgNBReconfComp
{
    oneof value
    {
        ResponseInformationSgNBReconfComp_SuccessItem success_SgNBReconfComp = 1;
        ResponseInformationSgNBReconfComp_RejectByMeNBItem reject_by_MeNB_SgNBReconfComp = 2;
    }
}

message ResponseInformationSgNBReconfComp_SuccessItem
{
    google.protobuf.BytesValue meNBtoSgNBContainer = 1;
    repeated ResponseInformationSgNBReconfComp_SuccessItemExtIEs iE_Extensions = 2;
}

message ResponseInformationSgNBReconfComp_SuccessItemExtIEs
{
}

message ResponseInformationSgNBReconfComp_RejectByMeNBItem
{
    Cause cause = 1;
    repeated ResponseInformationSgNBReconfComp_RejectByMeNBItemExtIEs iE_Extensions = 2;
}

message ResponseInformationSgNBReconfComp_RejectByMeNBItemExtIEs
{
}
